invisible_cities/cities/icaro.py

```diff
@@ -5,31 +5,51 @@
 
 
 """
-from operator import attrgetter
 
+import numpy as np
+import pandas as pd
 import tables as tb
 
-from .. reco         import tbl_functions as tbl
-from .. io.krmaps_io import write_krmap
-from .. io.krmaps_io import write_krevol
-from .. io.krmaps_io import write_mapinfo
+from .. reco                import tbl_functions as tbl
+from .. reco.corrections    import ASectorMap
+from .. reco.corrections    import maps_coefficient_getter
+from .. reco.corrections    import correct_geometry_
+from .. reco.corrections    import norm_strategy
+from .. reco.corrections    import get_normalization_factor
+from .. reco.corrections    import apply_all_correction
+from .. io.krmaps_io        import write_krmap
+from .. io.krmaps_io        import write_krevol
+from .. io.krmaps_io        import write_mapinfo
+from .. core                import fit_functions as fitf
+from .. core.core_functions import in_range
+from .. core.core_functions import shift_to_bin_centers
 
 from .. dataflow          import dataflow as fl
 from .. dataflow.dataflow import push
 from .. dataflow.dataflow import pipe
 
+from .. types.ic_types    import Measurement
+from .. types.ic_types    import MasksContainer
+
 from .  components import city
 from .  components import print_numberofevents
 from .  components import dst_from_files
 from .  components import quality_check
 from .  components import kr_selection
 from .  components import map_builder
 from .  components import add_krevol
+from .  components import print_every
+
+from typing import Tuple
+from typing import List
+from pandas import DataFrame
 
 
 @city
-def icaro(files_in, file_out, compression, detector_db, run_number,
-          bootstrap, quality_ranges, band_sel_params, map_params, krevol_params):
+def icaro(files_in, file_out, compression, event_range, print_mod,
```

Comment:
Why do you introduce event_range and print_mod? If I'm not wrong, the aim is that all the events are run in one interaction of the pipe. Therefore, those variables don't make sense in that context.

Reply:
Below there is:
```python
fl.slice(*event_range, close_all=True),
print_every(print_mod)                ,
```
so I added them to the list of arguments (as I saw in other cities). If we remove them, we should also change those lines I guess.

Reply:
In that regard, I think that the best example for the concept I mean is [Eutropia](https://github.com/next-exp/IC/blob/master/invisible_cities/cities/eutropia.py), that is also taking the whole range of events as a whole and doing a single iteration. Do you agree, @gonzaponte?

Reply:
IIRC the city _needs_ the parameter `event_range`. That's probably why _Eutropia_ takes it even though it doesn't use it (Although skimming through the code I didn't see why it is needed). But yes, in this case, we don't want to iterate over single events so the Eutropia example is a good one.

Reply:
Ok, I removed those lines and the `print_mod` argument for now.